Oxide.Core/RemoteConsole/RemoteConsole.cs

```diff
@@ -81,77 +81,126 @@ public void SendMessage(RemoteMessage message)
                     break;
 
-                case "chat":
-                    covalence.Server.Broadcast($"{config.ChatPrefix}: {message.Message}");
+                case "server.save":
+                case "save":
+                    covalence.Server.Save();
                     break;
 
-                case "players":
-                    SendMessage(RemoteMessage.CreateMessage(GetPlayerList(), 0, "players"));
+                case "ban":
+                case "banid":
+                case "global.ban":
+                case "global.banid":
+                    if (args.Length == 0)
+                        return;
+                    if (Interface.Call<bool?>("OnRconBan", args) != false)
+                        return;
+                    var banplayer = covalence.Players.FindPlayer(args[0]);
+                    if (banplayer != null)
+                        banplayer.Ban(string.Join(" ", args.Skip(1).ToArray()));
+                    else
+                        covalence.Server.Ban(args[0], string.Join(" ", args.Skip(1).ToArray()));
+                    break;
+
+                case "unban":
+                case "global.unban":
+                    if (args.Length == 0)
+                        return;
+                    if (Interface.Call<bool?>("OnRconUnban", args) != false)
+                        return;
+                    var unbanplayer = covalence.Players.FindPlayer(string.Join(" ", args));
+                    if (unbanplayer != null)
+                        unbanplayer.Unban();
+                    else
+                        covalence.Server.Unban(args[0]);
+                    break;
+
+                case "version":
+                case "server.version":
+                    var serverversion = covalence.Server.Version;
+                    var Game = covalence.Game;
+                    var oxideversion = OxideMod.Version;
+                    SendMessage(RemoteMessage.CreateMessage($"{Game} - {serverversion} Protocol {covalence.Server.Protocol} with OxideMod v{oxideversion.Major}.{oxideversion.Minor}.{oxideversion.Patch}"));
                     break;
 
                 default:
-                    SendMessage(RemoteMessage.CreateMessage("Unknown command"));
+                    covalence.Server.Command(cmd, args);
                     break;
             }
         }
 
         private class RconPlayer
         {
-            public string SteamId { get; private set; }
-            public string Name { get; private set; }
+            public string SteamID { get; private set; }
+            public string OwnerSteamID { get; private set; }
+            public string DisplayName { get; private set; }
+            public int Ping { get; private set; }
             public string Address { get; private set; }
+            public uint ConnectedSeconds { get; private set; }
+            public double VoiationLevel { get; private set; }
```

Comment:
Yes these are rust values but are necessary for this to work with current Rust Rcon Clients.